S3StorageWagon/src/main/java/com/gkatzioura/maven/cloud/s3/S3StorageRepository.java

```diff
@@ -98,6 +100,8 @@ public void copy(String resourceName, File destination, TransferProgress transfe
 
         final String key = resolveKey(resourceName);
 
+        destination.getParentFile().mkdirs();  // ensure location can be copied to
```

Comment:
Otherwise if the directory doesn't exist this fails, which doesn't seem right for recursive copy.